File: api/src/endpoint/repository/get_repository_commits.rs
use serde::{Deserialize, Serialize};

use super::{default_page, default_per_page, default_ref};
use crate::endpoint::Endpoint;
use crate::resource::repository::RepositoryCommitsResource;

pub struct GetRepositoryCommits;

impl Endpoint for GetRepositoryCommits {
    const PATH: &'static str = "/repository/{owner}/{repo}/commits";
    const METHOD: http::Method = http::Method::GET;

    type Request = GetRepositoryCommitsRequest;
    type Response = GetRepositoryCommitsResponse;
}

#[derive(Debug, Serialize, Deserialize)]
pub struct GetRepositoryCommitsRequest {
    #[serde(default = "default_ref")]
    pub ref_name: String,
    #[serde(default = "default_page")]
    pub page: u32,
    #[serde(default = "default_per_page")]
    pub per_page: u32,
}

pub type GetRepositoryCommitsResponse = RepositoryCommitsResource;

File: gitdot-api-ts/src/endpoint/repository/get-repository-commits.ts
import { z } from "zod";

import { RepositoryCommitsResource } from "../../resource";

export const GetRepositoryCommitsRequest = z.object({
  ref_name: z.string(),
  page: z.number().int(),
  per_page: z.number().int(),
});
export type GetRepositoryCommitsRequest = z.infer<
  typeof GetRepositoryCommitsRequest
>;

export const GetRepositoryCommits = {
  path: "/repository/{owner}/{repo}/commits",
  method: "GET",
  request: GetRepositoryCommitsRequest,
  response: RepositoryCommitsResource,
} as const;
export type GetRepositoryCommits = typeof GetRepositoryCommits;
